MediaBrowser.Common/Plugins/BasePlugin.cs

```diff
@@ -149,55 +166,54 @@ public TConfigurationType Configuration
                         }
                     }
                 }
+
                 return _configuration;
             }
-            protected set => _configuration = value;
-        }
 
-        private TConfigurationType LoadConfiguration()
-        {
-            var path = ConfigurationFilePath;
-
-            try
-            {
-                return (TConfigurationType)XmlSerializer.DeserializeFromFile(typeof(TConfigurationType), path);
-            }
-            catch
-            {
-                return (TConfigurationType)Activator.CreateInstance(typeof(TConfigurationType));
-            }
+            protected set => _configuration = value;
         }
 
         /// <summary>
-        /// Gets the name of the configuration file. Subclasses should override
+        /// Gets the name of the configuration file. Subclasses should override.
         /// </summary>
         /// <value>The name of the configuration file.</value>
         public virtual string ConfigurationFileName => Path.ChangeExtension(AssemblyFileName, ".xml");
 
         /// <summary>
-        /// Gets the full path to the configuration file
+        /// Gets the full path to the configuration file.
         /// </summary>
         /// <value>The configuration file path.</value>
         public string ConfigurationFilePath => Path.Combine(ApplicationPaths.PluginConfigurationsPath, ConfigurationFileName);
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="BasePlugin{TConfigurationType}" /> class.
+        /// Gets the plugin's configuration.
         /// </summary>
-        /// <param name="applicationPaths">The application paths.</param>
-        /// <param name="xmlSerializer">The XML serializer.</param>
-        protected BasePlugin(IApplicationPaths applicationPaths, IXmlSerializer xmlSerializer)
+        /// <value>The configuration.</value>
+        BasePluginConfiguration IHasPluginConfiguration.Configuration => Configuration;
+
+        /// <inheritdoc />
+        public void SetStartupInfo(Action<string> directoryCreateFn)
         {
-            ApplicationPaths = applicationPaths;
-            XmlSerializer = xmlSerializer;
+            // hack alert, until the .net core transition is complete
```

Comment:
heh, isn't .net core complete?..

Reply:
🤷‍♂ I just copy pasted this, no idea what this is about

Reply:
Looks like a place for further improvements...